addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,193 @@
+                            new HttpMessage(new org.apache.commons.httpclient.URI(TMP_URI, true));
+                    sendAndReceive(req_with_payload);
+                    if (req_with_payload.getResponseHeader().getStatusCode() == HttpStatusCode.OK) {
+                        newAlert()
+                                .setConfidence(Alert.RISK_INFO)
+                                .setName(
+                                        Constant.messages.getString(
+                                                "ascanalpha.forbiddenBypass.name"))
+                                .setDescription(
+                                        Constant.messages.getString(
+                                                "ascanalpha.forbiddenBypass.desc"))
+                                .setOtherInfo(TMP_URI)
+                                .setMessage(req_with_payload)
+                                .raise();
+                    }
+                }
+
+                // sending payloads in Headers.
+                for (String h : hpayloads) {
+                    String TMP_URI = SCHEMA + "://" + HOST + PATH;
+                    if (h.contains("X-Rewrite-URL")) {
+                        TMP_URI = SCHEMA + "://" + HOST + "/anything";
+                    } else if (h.contains("X-Original-URL")) {
+                        TMP_URI = SCHEMA + "://" + HOST + "/";
+                    } else if (h.contains("Referer")) {
+                        TMP_URI = SCHEMA + "://" + HOST + "/anything";
+                    }
+                    HttpMessage req_with_payload1 =
+                            new HttpMessage(new org.apache.commons.httpclient.URI(TMP_URI, true));
+                    req_with_payload1
+                            .getRequestHeader()
+                            .setHeader(h.split(":")[0], h.split(":")[1]);
+                    sendAndReceive(req_with_payload1);
+                    if (req_with_payload1.getResponseHeader().getStatusCode()
+                            == HttpStatusCode.OK) {
+                        System.out.println(req_with_payload1.getResponseHeader().getStatusCode());
+                        newAlert()
+                                .setConfidence(Alert.RISK_INFO)
+                                .setName(
+                                        Constant.messages.getString(
+                                                "ascanalpha.forbiddenBypass.name"))
+                                .setDescription(
+                                        Constant.messages.getString(
+                                                "ascanalpha.forbiddenBypass.desc"))
+                                .setOtherInfo(FULL_URL)
+                                .setMessage(req_with_payload1)
+                                .raise();
+                    }
+                }
+            }
+        } catch (IOException | URISyntaxException e) {
+            log.error(e.getMessage(), e);
```

Comment:
Should not be error, I/O errors are expected to happen.

Reply:
So what should I add here?

Reply:
I'd suggest reduce to debug or warn at most.